Feature/job application detail

ParimaSA opened the pull request:
### What's new
- Update company dashboard and create new page for job application management.
- I have changed the company dashboard to show the statistical information without clickable.
- Dashboard show all the job in the company which can be filtered by the status of the job.
- Job management page show the statistical information with the detail of each applicant and the button to manage their status.
- All the dashboard are responsive.

### Functional Acceptance Criteria

- Dashboard page show all job and be able to filter the data by the status.
- Only job with approved status can be linked to the management page.
- Management page show the data of each applicant and be able to change the status.

### What's next

- Integrate data with the backend API.
- Make the page responsive.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Testing
- [ ] Documentation update
- [ ] Other

Pat-7-626 commented:
<img width="460" height="134" alt="image" src="https://github.com/user-attachments/assets/25c820ef-cadb-4fd2-a6c8-79dff65d6e8f" />

So, this section `Location` from the form input will not be shown in the statistical information? (not a required change, just a question.)

ParimaSA commented:
> <img alt="image" width="460" height="134" src="https://private-user-images.githubusercontent.com/142660659/502978158-25c820ef-cadb-4fd2-a6c8-79dff65d6e8f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjExMTI5MTgsIm5iZiI6MTc2MTExMjYxOCwicGF0aCI6Ii8xNDI2NjA2NTkvNTAyOTc4MTU4LTI1YzgyMGVmLWNhZGItNGZkMi1hNmM4LTc5ZGZmNjVkNmU4Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyMlQwNTU2NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzA3NDJhMGIxMThmYmQ5YjI4MGYxYmU1ODcwOTAwYmM3NjliYzc0NDM2ZmRlMjk2NDg1M2Q3NzQwMDk1ODE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UfVdrgjwPQsXLkDui0ogA8cd7ZGUWq-tdhhQF-51eoE">
> 
> So, this section `Location` from the form input will not be shown in the statistical information? (not a required change, just a question.)

I’ve updated the type and added the location field to the applicant card.

Pat-7-626 reviewed (approved):
Then, I approve.

ParimaSA commented:
I’ve updated the behavior so that both Approve and Reject buttons remain visible when the user hasn’t saved their changes yet, allowing them to easily switch status before saving.

> However, when an applicant is approved/rejected for a job, all other jobs that the student has applied to for the company is also approved/rejected.

This happens because of the mock data, most entries except for those in the second job, already have their status set to approved or rejected. Currently, the save functionality only logs the changes without actually updating the data, so it doesn’t affect other application records. Additionally, I’ve updated the mock job data to correctly reflect the applicant numbers.

Lastly, I’ve added a confirmation modal that summarizes all changes and asks for confirmation before the updates are permanently saved.
